src/app/components/pages/IsaacCompetition/EntryForm/useReserveUsersOnCompetition.ts

```diff
@@ -0,0 +1,21 @@
+import { api } from "../../../../services";
+import { showErrorToast, showSuccessToast } from "../../../../state/actions/popups";
+import { AppDispatch } from "../../../../state";
+
+export const useReserveUsersOnCompetition = (dispatch: AppDispatch) => {
+  const reserveUsersOnCompetition = async (
```

Comment:
you could just call in the `useAppDispatch` hook in here, so that you don't need to pass it from the main component. It's part of the underlying function so it makes sense to do it :)